### PR TITLE
move call of suicide stalemate

### PR DESCRIPTION
**Pull request by alwey**
I suggest to move the suicide_stalemate handling a few lines down in search.cpp. This might have been a glitch while merging.

### CODE REVIEW

**Comment by ianfab** [comment]
Thanks, I have somehow put it in the wrong place.

File: src/search.cpp [hunk 1]
```diff
@@ -1319,14 +1319,14 @@ namespace {
 #endif
 #ifdef HORDE
         if (pos.is_horde() && pos.is_horde_loss())
-            bestValue = excludedMove ? alpha
-            : pos.is_suicide() ? pos.suicide_stalemate(ss->ply, DrawValue[pos.side_to_move()])
-            : mate_in(ss->ply+1);
+            bestValue = excludedMove ? alpha : mate_in(ss->ply+1);
```
Comment (ianfab) [inline src/search.cpp]: I think this should be `mated_in(ss->ply)` instead of `mate_in(ss->ply+1)`.

Comment (alwey) [inline src/search.cpp]: Affirmative, must be `mated_in(ss->ply)`, This is _my_ glitch. Sorry and thank you. Will update accordingly.

**Comment by alwey** [comment]
Second try.
@ianfab: I would also be interested in SF playing original Losing Chess (Vinciperdi in Italian and Qui perd gagne in French and Räuberschach in German: variant  `lc1`,` losing` or `anti`) which is giveaway sans castling.

**Comment by ianfab** [comment]
@alwey It should be easy to add losing chess as a subvariant. However, SF already plays it when you provide an FEN without castling rights in giveaway. Anyway, if @ddugovic is ok with it, I could write the code and open a PR, but I would first have to know how to name the subvariant within the code, since the name losing could be misleading in some contexts. Is there a naming convention for this variant in the xboard protocol? I so far only know of giveaway, suicide, and losers being implemented in engines.

**Comment by alwey** [comment]
@ianfab: indeed, thank you. I already used the method of FEN with removed castling rights and it worked with `giveaway ` variant some weeks ago: I replaced the `giveaway` variant name with `anti` and then with `antichess`. I needed this for an extension I wrote for cutechess / cutechess-cli. I only have SF vs SF games so far, because I do not have any other engine supporting antichess / LC1. I downloaded some `antichess` games from lichess.org in order to test the import function of cutechess with my code.

**Comment by ddugovic** [comment]
I'm okay with creating another branch for vinciperdi, although I'm not sure that there's a compelling reason to implement it in the master branch since aside from lichess, no popular site seems to support the variant.

**Comment by alwey** [comment]
@ddugovic : Thank you. Variant support might be a hen-and-egg problem.

**Comment by ddugovic** [comment]
Possibly so since vinciperdi hasn't been popular in decades, if ever.  Not sure why lichess avoided popular ICS/FICS/USCL rules and accidentally popularize it.

**Comment by alwey** [comment]
I am no expert. I have heard or read from some strong Losing Chess players who definitively preferred the old ruleset (LC1, international rules). If literature exists, it will probably use these rules.

**Comment by ddugovic** [comment]
Is there any position in any literature where omission of castling rights makes a difference?  Honestly I'm curious since perhaps we can convince ICC/FICS to adopt LC1.

**Comment by alwey** [comment]
I do not think it is necessary to convince them. I am asking myself whether the distinction between these variants has been programmer driven, user driven or driven by legal considerations of chess server maintainers.

**Comment by ddugovic** [comment]
My consideration was that with respect to ICC, FICS, WinBoard, and other chess engines Stockfish is not deficient; and that the user interface can explicitly specify whether or not castling is supported.

Similarly you don't see explicit support for no-castle variant because the user interface can already specify a position including castling rights.

**Comment by alwey** [comment]
No-castle is a good example.

**Pull request merged**